engine/execution/rpc/engine.go

```diff
@@ -164,7 +164,8 @@ func (h *handler) ExecuteScriptAtBlockID(
 
 	value, err := h.engine.ExecuteScriptAtBlockID(ctx, req.GetScript(), req.GetArguments(), blockID)
 	if err != nil {
-		return nil, status.Errorf(codes.Internal, "failed to execute script: %v", err)
+		// return code 3 as this passes the litmus test in our context
+		return nil, status.Errorf(codes.InvalidArgument, "failed to execute script: %v", err)
```

Comment:
maybe use `status.Code(err)` instead of hardcoding to `codes.InvalidArgument` in case the underlying implementation of `ExecuteScriptAtBlockID` changes

Reply:
But wouldn't that bound error type returned from engine to implement gRPC specific interface therefore bounding it to types?

Reply:
yea, after looking further, this doesn't make sense.